core/modules/book/book.module

```diff
@@ -1243,3 +1249,41 @@ function book_menu_subtree_data($link) {
 
   return $tree[$cid];
 }
+
+/**
+ * Invalidates cached data relating to book_cache.
+ *
+ * @param $cid
+ *   (optional) Cache ID of the record to clear from the private update module
+ *   cache. If empty, all records will be cleared from the table except fetch
+ *   tasks. Defaults to NULL.
+ * @param $wildcard
+ *   (optional) If TRUE, cache IDs starting with $cid are deleted in addition to
+ *   the exact cache ID specified by $cid. Defaults to FALSE.
+ */
+function _book_cache_clear($cid = NULL, $wildcard = FALSE) {
+  if (empty($cid)) {
+    cache('book')->delete($cid);
+  }
+  else {
+    cache('book')->deletePrefix($cid);
+  }
```

Comment:
These conditions don't seem to match the parameters. $wildcard is never checked and if $cid is empty, it should be a full table flush `cache('book')->flush()`. So there needs to at least be another condition check here.